test/test_ops.py

```diff
@@ -687,6 +687,29 @@ def test_composite_compliance(self, device, dtype, op):
             kwargs = sample.kwargs
             _check_composite_compliance(op, args, kwargs)
 
+    @onlyCPU
+    @ops(op_db, allowed_dtypes=(torch.float,))
+    def test_floating_inputs_are_differentiable(self, device, dtype, op):
```

Comment:
If this is just enforced and there is not way to make some inputs not require gradients, why don't we just set requires grad to True whatever the user returned?

Reply:
We want to be able to skip/xfail this for some operations, when the support is limited.

Reply:
Don't want a way for the user to specify that things explicitly shouldn't require gradients?
We could have a `mark_non_differentiable(some_tensor)` that users can use during the input creation.

And in that method we could set `some_tensor._not_requires_grad_on_purposes` and have this new check look for this field on the given Tensor?

Reply:
@mruberry preferred this test over marking individual tensors.